examples/js/loaders/OBJLoader.js

```diff
@@ -505,54 +515,114 @@ THREE.OBJLoader.prototype = {
 
 			} else if ( lineFirstChar === "f" ) {
 
+				// f vertex/uv/normal vertex/uv/normal vertex/uv/normal
+				// 0                        1    2    3    4    5    6    7    8    9   10         11         12
+				// ["f 1/1/1 2/2/2 3/3/3", "1", "1", "1", "2", "2", "2", "3", "3", "3", undefined, undefined, undefined]
+
 				if ( ( result = this.regexp.face_vertex_uv_normal.exec( line ) ) !== null ) {
 
-					// f vertex/uv/normal vertex/uv/normal vertex/uv/normal
-					// 0                        1    2    3    4    5    6    7    8    9   10         11         12
-					// ["f 1/1/1 2/2/2 3/3/3", "1", "1", "1", "2", "2", "2", "3", "3", "3", undefined, undefined, undefined]
+					var rawdata = /^f\s+/.exec( line );
+					var data = line.slice( rawdata[ 0 ].length );
+					var vuns = [];
+					var splitData = data.split( ' ' );
 
-					state.addFace(
-						result[ 1 ], result[ 4 ], result[ 7 ], result[ 10 ],
-						result[ 2 ], result[ 5 ], result[ 8 ], result[ 11 ],
-						result[ 3 ], result[ 6 ], result[ 9 ], result[ 12 ]
-					);
+					for ( var j = 0; j < splitData.length; j ++ ) {
+
+						vuns = vuns.concat( splitData[ j ].split( '/' ) );
+
+					}
+
+					var vs = [];
+					var us = [];
+					var ns = [];
+					for ( var j = 0; j < vuns.length / 3; j ++ ) {
+
+						vs.push( vuns[ 3 * j ] );
+						us.push( vuns[ 3 * j + 1 ] );
+						ns.push( vuns[ 3 * j + 2 ] );
+
+					}
+
+					state.addFace( vs, us, ns );
```

Comment:
As far as I can tell from my earlier testing the slow parts in the loader are array manipulations. This adds a bunch more. Just wondering about the loader perf impact here.